### DISCUSSION TITLE
Cargo.lock in repository

### DISCUSSION THREAD

**Discussion by jaybosamiya** [verus-lang/verus#368] [General]
At some point, we should consider maintaining a `Cargo.lock` in the repository.

It was originally `.gitignore`d in 2021: https://github.com/verus-lang/verus/commit/d4ded8f7cd9c52ecb277ec260f4b12a52fc7239f

https://github.com/verus-lang/verus/pull/360 tried to (re)introduce it (including some rationale for why we should be including `Cargo.lock`). Since non-trivial changes were soon to be made (to move off the rustc fork), adding `Cargo.lock` right now as decided against.

Likely, once we have moved off of the rustc fork, we should revisit this and lock our dependencies to make longer term maintenance easier.

If any more reasons (other than the rustc fork move) exist to _not_ keep the `Cargo.lock` around, it would be good to hear them, thus this discussion thread. For reasons to maintain `Cargo.lock`, some are already listed on #360.

**Comment by utaal** [comment]
Being added by @jaybosamiya in https://github.com/verus-lang/verus/pull/669.
